irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -29,44 +32,99 @@ namespace iroha {
+
       log_->info("transactions in proposal: {}",
                  proposal.transactions().size());
-      auto checking_transaction = [](const auto &tx, auto &queries) {
-        return bool(queries.getAccount(tx.creatorAccountId()) |
-                    [&](const auto &account) {
-                      // Check if tx creator has account and has quorum to
-                      // execute transaction
-                      return boost::size(tx.signatures()) >= account->quorum()
-                          ? queries.getSignatories(tx.creatorAccountId())
-                          : boost::none;
-                    }
-                    |
-                    [&](const auto &signatories) {
-                      // Check if signatures in transaction are account
-                      // signatory
-                      return signaturesSubset(tx.signatures(), signatories)
-                          ? boost::make_optional(signatories)
-                          : boost::none;
-                    });
+      auto checking_transaction = [this](const auto &tx, auto &queries) {
+        return expected::Result<void, std::string>(
+            [&]() -> expected::Result<
+                         std::shared_ptr<shared_model::interface::Account>,
+                         std::string> {
+              // Check if tx creator has account
+              auto account = queries.getAccount(tx.creatorAccountId());
+              if (account) {
+                return expected::makeValue(*account);
+              }
+              return expected::makeError(
+                  (boost::format("stateful validator error: could not fetch "
+                                 "account with id %s")
+                   % tx.creatorAccountId())
+                      .str());
+            }() |
+                [&](const auto &account)
+                      -> expected::Result<
+                             std::vector<
+                                 shared_model::interface::types::PubkeyType>,
+                             std::string> {
+              // Check if account has signatories and quorum to execute
+              // transaction
+              if (boost::size(tx.signatures()) >= account->quorum()) {
+                auto signatories =
+                    queries.getSignatories(tx.creatorAccountId());
+                if (signatories) {
+                  return expected::makeValue(*signatories);
+                }
+                return expected::makeError(
+                    (boost::format("stateful validator error: could not fetch "
+                                   "signatories of "
+                                   "account %s")
+                     % tx.creatorAccountId())
+                        .str());
+              }
+              return expected::makeError(
+                  (boost::format(
+                       "stateful validator error: not enough "
+                       "signatures on account %s; account's quorum %d, "
```

Comment:
Not sure if I understood the message correctly, but maybe "not enough signatures in transaction" is more straightforward? Because the error appears when `tx.sigs < acc.quorum`.